### ISSUE TITLE
SIGABRT on output change

### ISSUE THREAD

**Issue by flub** [riverwm/river#785]
It seems like occasionally I get river to panic, I think triggered by changes to my external displays.  I looked at a coredump that maybe points to somewhere in SeatStatus.zig but I don't know zig so not sure what can be the cause here:

```
Core was generated by `/home/flub/.local/bin/river'.
Program terminated with signal SIGABRT, Aborted.
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at ./nptl/pthread_kill.c:44
44	./nptl/pthread_kill.c: No such file or directory.
[Current thread is 1 (Thread 0x7f89cf805980 (LWP 12574))]
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at ./nptl/pthread_kill.c:44
#1  0x00007f89cfc32d2f in __pthread_kill_internal (signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:78
#2  0x00007f89cfbe3ef2 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007f89cfbce472 in __GI_abort () at ./stdlib/abort.c:79
#4  0x0000000000283f66 in os.abort () at /home/flub/src/zig-linux-x86_64-0.10.1/lib/std/os.zig:560
#5  0x000000000028481b in debug.panicImpl (trace=<optimized out>, first_trace_addr=..., msg=...) at /home/flub/src/zig-linux-x86_64-0.10.1/lib/std/debug.zig:384
#6  0x00000000002340f7 in builtin.default_panic (msg=..., error_return_trace=0x0, ret_addr=...) at /home/flub/src/zig-linux-x86_64-0.10.1/lib/std/builtin.zig:839
#7  0x0000000000252140 in Seat.setFocusRaw (self=0xc138d0, new_focus=...) at SeatStatus.zig:70
#8  0x0000000000279878 in Output.arrangeLayers (self=0xfeae70, target=<optimized out>) at Output.zig:315
#9  0x0000000000282289 in LayerSurface.handleMap () at LayerSurface.zig:103
#10 wayland_server_core.Listener(*types.layer_shell_v1.LayerSurfaceV1).setNotify__struct_11597.wrapper (listener=0x10402b0, data=0xed86c0)
    at /home/flub/src/river/zig-cache/zig-wayland/wayland_server_core.zig:520
#11 0x00007f89cff8890c in wl_signal_emit_mutable () at /usr/lib/x86_64-linux-gnu/libwayland-server.so.0
#12 0x00007f89cfdfbc1b in  () at /usr/lib/x86_64-linux-gnu/libwlroots.so.11
#13 0x00007f89cfa68f7a in  () at /lib/x86_64-linux-gnu/libffi.so.8
#14 0x00007f89cfa6840e in  () at /lib/x86_64-linux-gnu/libffi.so.8
#15 0x00007f89cfa68b0d in ffi_call () at /lib/x86_64-linux-gnu/libffi.so.8
#16 0x00007f89cff8c981 in  () at /usr/lib/x86_64-linux-gnu/libwayland-server.so.0
#17 0x00007f89cff87bda in  () at /usr/lib/x86_64-linux-gnu/libwayland-server.so.0
#18 0x00007f89cff8a89a in wl_event_loop_dispatch () at /usr/lib/x86_64-linux-gnu/libwayland-server.so.0
#19 0x00007f89cff88365 in wl_display_run () at /usr/lib/x86_64-linux-gnu/libwayland-server.so.0
#20 0x0000000000233c0d in main.main () at main.zig:132
#21 start.callMain () at /home/flub/src/zig-linux-x86_64-0.10.1/lib/std/start.zig:614
#22 start.initEventLoopAndCallMain () at /home/flub/src/zig-linux-x86_64-0.10.1/lib/std/start.zig:548
#23 start.callMainWithArgs () at /home/flub/src/zig-linux-x86_64-0.10.1/lib/std/start.zig:498
#24 main (c_argc=<optimized out>, c_argv=<optimized out>, c_envp=<optimized out>) at /home/flub/src/zig-linux-x86_64-0.10.1/lib/std/start.zig:513
```

This is compiled from tag v0.2.3.  Where `SeatStatus.zig:70` is:
```
68 pub fn sendFocusedView(self: Self) void {
69     const title: [*:0]const u8 = if (self.seat.focused == .view)
70         self.seat.focused.view.getTitle() orelse ""
71     else
72         "";
73     self.seat_status.sendFocusedView(title);
74 }
```

**Comment by ifreund** [comment]
Did this happen while the session was locked by any chance?

**Comment by flub** [comment]
Yes you're right, the session was locked.  Sorry to forget to add this.

**Comment by ifreund** [comment]
No problem! I think this should be fixed by e1ab51f26071958bbeb2d0c637d3d00eb4326fdf. Let me know if you hit it again with that commit.

And thanks for the stack trace, I love bug reports like this :)

**Comment by flub** [comment]
Thanks for the quick fix!  Did a small amount of testing and so far
seems like it fixed. it.
